[Term Entry] Python Built-in Functions property() (#2743)
Codecademy/docs: content/python/concepts/built-in-functions/terms/property/property.md

---
Title: 'property()'

> Description: 'Declares property to access class attributes.'

Reviewer: ```suggestion
Description: 'Declares a range of functions to manipulate class attributes.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Classes'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`property()`** function is a built-in Python function used to create properties by defining getter, setter and deleter for class attributes. It returns the object of property class that can be used as decorator or assigned to a class attribute.

Reviewer: ```suggestion
The **`property()`** function is a built-in Python function used to define methods that get, set, and delete class attributes. It can be used as a [decorator](https://www.codecademy.com/resources/docs/python/decorators) or assigned to a class attribute.
```

## Syntax

```pseudo
property(fget, fset, fdel, doc)
```

The `property()` function requires four parameters:

> - `fget`: function to get the value of the `attribute`.
> - `fset`: function to set the value of the `attribute`.
> - `fdel`: function to delete the value of the `attribute`.
> - `doc`: string that creates `docstring` for the `attribute`.

Reviewer: ```suggestion
- `fget`: function to get the value of the attribute.
- `fset`: function to set the value of the attribute.
- `fdel`: function to delete the value of the attribute.
- `doc`: string that creates `docstring` for the attribute.
```

> > **Note:** In case no arguments are given then the function returns a base property attribute without any getter, setter or deleter.

Reviewer: ```suggestion
> **Note:** If no arguments are given then the function returns a base property attribute without any getter, setter or deleter.
```

## Example

The following example will demonstrate the use of `property()` function using the `Pet` class:

```py
class Pet:
    def __init__(self, name):
        self._name = name

    def get_name(self):
        return self._name

    def set_name(self, name):
        self._name = name

    def del_name(self):
        del self._name

    name = property(get_name, set_name, del_name)

# Create an instance of Pet class
dog = Pet("Bruno")

# Access the name attribute using the property
print(dog.name)

# Set a new value for the name attribute
dog.name = "Mars"

# Access the updated name attribute
print(dog.name)

# Delete the name attribute
del dog.name

> # Accessing the radius after deletion will raise an AttributeError

Reviewer: ```suggestion
```

```

The code results in the following output:

```shell
Bruno
Mars
```

> ## Codebyte Example

Reviewer: This example is a very close copy of an example on another site. All docs content must be original, we do not permit plagiarism. Please revise or delete this example.

Author: I am pretty sure I came up with it myself, still apologies. Which example you want me to update the example 1 or the codebyte example 2?
